pkg/debugapi/balances.go

```diff
@@ -0,0 +1,72 @@
+package debugapi
+
+import (
+	"sort"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/gorilla/mux"
+
+	"net/http"
+)
+
+type balanceResponse struct {
+	Peer    string `json:"peer"`
+	Balance int64  `json:"balance"`
+}
+
+type balancesResponse struct {
+	Balances []balanceResponse `json:"balances"`
+}
+
+func (s *server) balancesHandler(w http.ResponseWriter, r *http.Request) {
+
+	balances, err := s.Accounting.Balances()
+
+	if err != nil {
+		// TODO: Do we need to return an explicit error?
+		s.Logger.Debugf("debug api: balances: %v", err)
+	}
+
+	var balResponses []balanceResponse
+
+	keys := make([]string, 0, len(balances))
+	for k := range balances {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		balResponses = append(balResponses, balanceResponse{
+			Peer:    k,
+			Balance: balances[k],
+		})
+	}
+
+	jsonhttp.OK(w, balancesResponse{Balances: balResponses})
+
+}
+
+func (s *server) peerBalanceHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO: Currently, we do not check the length of the hex address (such as prepending zeroes), should we?
+	peer, err := swarm.ParseHexAddress(mux.Vars(r)["peer"])
+	if err != nil {
+		s.Logger.Debugf("debug api: balances peer: parse peer address: %v", err)
+		jsonhttp.BadRequest(w, "malformed peer address")
+		return
+	}
+
+	balance, err := s.Accounting.Balance(peer)
+
+	if err != nil {
+		s.Logger.Debugf("debug-api: balances peer: get peer balance: %v", err)
+		jsonhttp.InternalServerError(w, err)
```

Comment:
I've looked at other api calls and it seems usually specific error message are never returned through http, only logged (twice).

Errors in the other apis are always handled like this:
```go
s.Logger.Debugf("file upload: file store, file %q: %v", fileName, err)
s.Logger.Errorf("file upload: file store, file %q", fileName)
jsonhttp.InternalServerError(w, "could not store file data")
```

Reply:
yes, that seems to be the case. 

i'm not a huge fan of this logging twice business, so i want to bring this up with @janos or @acud when they come back. i also [discussed it](https://github.com/ethersphere/bee/pull/447#discussion_r458803331) with @zbiljic in another PR.

not returning `err` info in the HTTP response is OK though.

Reply:
let's stick with the log twice convention for now. in any case, it will be better that we _don't_ leak developer/internal errors outside the application e.g. on api returned errors

Reply:
@mortelli @ralph-pichler reasons behind such logging is explained in https://github.com/ethersphere/bee/blob/master/CODING.md#logging.

Yes, http response should not expose arbitrary internal information.